evm/spec/cpulogic.tex

```diff
@@ -0,0 +1,170 @@
+\section{CPU logic}
+\label{cpulogic}
+
+The CPU is in charge of coordinating the different STARKs, proving the correct execution of the instructions it reads and guaranteeing
+that the final state of the EVM corresponds to the starting state after executing the input transaction. All design choices were made 
+to make sure these properties can be adequately translated into constraints of degree at most 3 while minimizing the size of the different
+table traces (number of columns and number of rows).
+
+In this section, we will detail some of these choices.
+
+\subsection{Kernel}
+The kernel is in charge of the proving logic. This section aims at providing a high level overview of this logic. For details about any specific part of the logic, one can consult the various ``asm'' files in the \href{https://github.com/0xPolygonZero/plonky2/tree/main/evm/src/cpu/kernel}{``kernel'' folder}.
+
+We prove one transaction at a time. These proofs can later be aggregated recursively to prove a block. Proof aggregation is however not in the scope of this section. Here, we assume that we have an initial state of the EVM, and we wish to prove that a single transaction was correctly executed, leading to a correct update of the state.
+
+Since we proceed one transaction at a time, a few intermediary values need to be provided by the prover. Indeed, to prove that the registers in the EVM state are currently updated, we need to have access to their initial values. When aggregating proofs, we can also constrain those values to match from one transaction to the next. Let us consider the example of the transaction number. Let $n$ be the number of transactions executed so far in the current block. If the current proof is not a dummy one (we are indeed executing a transaction), then the transaction number should be updated: $n := n+1$. Otherwise, the number reamins unchanged. We can easily constrain this update. When aggregating the previous transaction proof ($lhs$) with the current one ($rhs$), we also need to check that the output transaction number of $lhs$ is the same as the input transaction number of $rhs$. 
```

Comment:
```suggestion
Since we process one transaction at a time, a few intermediary values need to be provided by the prover. Indeed, to prove that the registers in the EVM state are correctly updated, we need to have access to their initial values. When aggregating proofs, we can also constrain those values to match from one transaction to the next. Let us consider the example of the transaction number. Let $n$ be the number of transactions executed so far in the current block. If the current proof is not a dummy one (we are indeed executing a transaction), then the transaction number should be updated: $n := n+1$. Otherwise, the number remains unchanged. We can easily constrain this update. When aggregating the previous transaction proof ($lhs$) with the current one ($rhs$), we also need to check that the output transaction number of $lhs$ is the same as the input transaction number of $rhs$. 
```